### PR TITLE
Initial try

### PR DESCRIPTION
**Pull request by vipul1409**
MultiThreadedFlatMapFunction provides an interface to execute execute flatmap using a set of threads. We can specify number of threads in the Constructor.

Usage flatMap(new MultiThreadedFlatMapFunction<String, Tuple2<String, Integer>>(new Tokenizer(),10)). Refer https://github.com/apache/flink/pull/2922/files#diff-27952c7e65576c8436355775051d8abb for usage example.

Please let me know if more details are needed.

### CODE REVIEW

File: flink-contrib/flink-streaming-contrib/src/test/java/org/apache/flink/contrib/streaming/MultiThreadedFlatMapFunctionTest.java [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+            System.out.println("Use --input to specify file input.");
+            // get default test text data
+            text = env.fromElements(WordCountData.WORDS);
+        }
+
+        DataStream<Tuple2<String, Integer>> counts =
+                // split up the lines in pairs (2-tuples) containing: (word,1)
+                //text.flatMap(new Tokenizer()).keyBy(0).sum(1);
+                text.flatMap(new MultiThreadedFlatMapFunction<String, Tuple2<String, Integer>>(new Tokenizer(),10)).
+                        returns(new TypeHint<Tuple2<String, Integer>>(){}).keyBy(0).sum(1);
```
Comment (vipul1409) [inline flink-contrib/flink-streaming-contrib/src/test/java/org/apache/flink/contrib/streaming/MultiThreadedFlatMapFunctionTest.java]: Can I avoid return call somehow?

File: flink-contrib/flink-streaming-contrib/src/main/java/org/apache/flink/contrib/streaming/MultiThreadedFlatMapFunction.java [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+                Collector<OUT> collector = new ListCollector<OUT>(result);
+                flatMapFunction.flatMap(value, collector);
+                return result;
+            }
+        });*/
+
+        futures.add(flatMapExecutors.submit(new Callable<ArrayList<OUT>>() {
+            @Override
+            public ArrayList<OUT> call() throws Exception {
+                ArrayList<OUT> result = new ArrayList<OUT>();
```
Comment (vipul1409) [inline flink-contrib/flink-streaming-contrib/src/main/java/org/apache/flink/contrib/streaming/MultiThreadedFlatMapFunction.java]: As each call may take significant time we need to ensure that all the results objects of active threads can stay in memory.

File: flink-contrib/flink-streaming-contrib/src/main/java/org/apache/flink/contrib/streaming/MultiThreadedFlatMapFunction.java [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+                Collector<OUT> collector = new ListCollector<OUT>(result);
```
Comment (vipul1409) [inline flink-contrib/flink-streaming-contrib/src/main/java/org/apache/flink/contrib/streaming/MultiThreadedFlatMapFunction.java]: Any suggestions on which collector would be a good choice over here. I am not aware of all the options. Is there any document available for different collectors.

**Comment by zentol** [comment]
Please include a description of th PR in the Title and main post; we can't review it like this.

**Comment by vipul1409** [comment]
Closing pull request as this is  not needed.

**Pull request closed without merging**